Source/LanderLads/SpaceshipPawn.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019 Dulce Studios. All Rights Reserved.
+
+#include "SpaceshipPawn.h"
+
+#include "UObject/ConstructorHelpers.h"
+
+// Sets default values
+ASpaceshipPawn::ASpaceshipPawn()
+{
+ 	// Set this pawn to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
+	PrimaryActorTick.bCanEverTick = true;
+
+	this->RootComponent = this->CreateDefaultSubobject<USceneComponent>(TEXT("DefaultSceneRoot"));
+	this->SpaceshipStaticMeshComponent = this->CreateDefaultSubobject<USpaceshipStaticMeshComponent>(
+		TEXT("SpaceshipStaticMeshComponent"));
+
+	this->SpaceshipStaticMeshComponent->AttachToComponent(
+		this->GetRootComponent(), FAttachmentTransformRules::KeepRelativeTransform);
+
+	this->ResultantForceVector = FVector(0);
+
+	static ConstructorHelpers::FObjectFinder<UCurveFloat> CurveAsset(
+		TEXT("CurveFloat'/Game/Misc/ThrusterJerk.ThrusterJerk'"));
+	this->CurveFloat = CurveAsset.Object;
+}
+
+// Called when the game starts or when spawned
+void ASpaceshipPawn::BeginPlay()
+{
+	Super::BeginPlay();
+
+	this->SpaceshipStaticMeshComponent->SetNotifyRigidBodyCollision(true);
+	this->SpaceshipStaticMeshComponent->OnComponentHit.AddDynamic(this, &ASpaceshipPawn::OnCompHit);
+}
+
+// Called every frame
+void ASpaceshipPawn::Tick(float DeltaTime)
+{
+	Super::Tick(DeltaTime);
+
+	this->GetSpaceshipStaticMeshComponent()-> AddForce(ResultantForceVector);
+}
+
+// Called to bind functionality to input
+void ASpaceshipPawn::SetupPlayerInputComponent(UInputComponent* PlayerInputComponent)
+{
+	Super::SetupPlayerInputComponent(PlayerInputComponent);
+
+	check(PlayerInputComponent);
+
+	PlayerInputComponent->BindAction("MoveUp", EInputEvent::IE_Pressed, this, &ASpaceshipPawn::MoveUpButtonPressed);
+	PlayerInputComponent->BindAction("MoveDown", EInputEvent::IE_Pressed, this, &ASpaceshipPawn::MoveDownButtonPressed);
+	PlayerInputComponent->BindAction("MoveRight", EInputEvent::IE_Pressed, this, &ASpaceshipPawn::MoveRightButtonPressed);
+	PlayerInputComponent->BindAction("MoveLeft", EInputEvent::IE_Pressed, this, &ASpaceshipPawn::MoveLeftButtonPressed);
+	PlayerInputComponent->BindAction("MoveForward", EInputEvent::IE_Pressed, this, &ASpaceshipPawn::MoveForwardButtonPressed);
+	PlayerInputComponent->BindAction("MoveBackward", EInputEvent::IE_Pressed, this, &ASpaceshipPawn::MoveBackwardButtonPressed);
```

Comment:
Why bind action if the input result works well with axis?

Reply:
Action binding is primarily for toggling the thrusters 😃